### PR TITLE
Ensure difficulty/hash/epoch overrides change the `ChainSpec`

### PR DESCRIPTION
**Pull request by paulhauner**
## Issue Addressed

NA

## Proposed Changes

Ensures the difficulty/hash/epoch overrides change the `ChainSpec`. This will result in changes to the `/eth/v1/config/spec` when the overrides are used, I *think* this is the correct behaviour (@ajsutton agrees, at least).

## Additional Info

NA

### CODE REVIEW

File: consensus/types/src/chain_spec.rs [hunk 1]
```diff
@@ -605,6 +609,12 @@ pub struct Config {
     #[serde(default)]
     pub preset_base: String,
 
+    #[serde(with = "eth2_serde_utils::quoted_u256")]
+    pub terminal_total_difficulty: Uint256,
+    pub terminal_block_hash: Hash256,
+    #[serde(with = "eth2_serde_utils::quoted_u64")]
```
Comment (realbigsean) [inline consensus/types/src/chain_spec.rs]: `Epoch` uses `quoted_u64` by default, so this shouldn't  be necessary

File: lighthouse/src/main.rs [hunk 2]
```diff
@@ -165,6 +165,45 @@ fn main() {
...
+            Arg::with_name("terminal-total-difficulty-override")
+                .long("terminal-total-difficulty-override")
+                .value_name("INTEGER")
+                .help("Used to coordinate manual overrides to the TERMINAL_TOTAL_DIFFICULTY parameter. \
+                       This flag should only be used if the user has a clear understanding that \
+                       the broad Ethereum community has elected to override the terminal difficulty. \
+                       Incorrect use of this flag will cause your node to experience a consensus
+                       failure. Be extremely careful with this flag.")
+                .takes_value(true)
+                .global(true)
```
Comment (realbigsean) [inline lighthouse/src/main.rs:179]: This is going to make these flags available in `lighthouse account_manager` which is a little strange.  But I think this is fixable once we upgrade to `clap v3` (v3 is still in beta but I implemented the upgrade here: https://github.com/sigp/lighthouse/pull/2808)

Conversation about why it's not yet fixable:
https://github.com/clap-rs/clap/issues/1183

File: lighthouse/src/main.rs [hunk 3]
```diff
@@ -165,6 +165,45 @@ fn main() {
...
+        )
+        .arg(
+            Arg::with_name("terminal-block-hash-override")
+                .long("terminal-block-hash-override")
+                .value_name("TERMINAL_BLOCK_HASH")
+                .help("Used to coordinate manual overrides to the TERMINAL_BLOCK_HASH parameter. \
+                       Accepts a 256-bit decimal integer (not a hex value). \
```
Comment (realbigsean) [inline lighthouse/src/main.rs]: Seems like this part of the description should be in the `terminal-total-difficulty-override` description?

**Review by realbigsean — CHANGES_REQUESTED** [review]
Nice! Just had a couple comments and nits.

General question: 

Since we check for complete spec compatibility, both the BN and VC will need the override flags to function. It seems like it'd be very difficult to actually socially coordinate this with things like Infura and BN fallbacks.  Is there away to go about this without having strict BN <> VC compatibility on these fields?

**Comment by paulhauner** [comment]
> Since we check for complete spec compatibility, both the BN and VC will need the override flags to function. It seems like it'd be very difficult to actually socially coordinate this with things like Infura and BN fallbacks. Is there away to go about this without having strict BN <> VC compatibility on these fields?

This is a good question. There was some discussion on the merge tech chat earlier this week. It seemed to get a consensus from clients that we *should* change this value. These values are a last resort and they're critical to consensus, since setting them wrong can lead to chain splits. So, if we do need to use them then it's going to be a massive coordination effort anyway. Notably, the LH VC only warns if the spec doesn't match, it doesn't error and exit anymore. So, I was thinking it's probably nice to have a little warning around if it turns out we need to do a big coordination effort on these.

File: lighthouse/src/main.rs (hunk 2 above)
Comment (paulhauner) [reply lighthouse/src/main.rs:179]: Yeah, it's indeed a bit unfortunate. However, I think I'm OK with that. The ability to ensure a consistent spec across all components is worth the occasional weirdness. With this setup, at least the `account_manager` has access to a correct spec, if it turns out it needs it later.

File: lighthouse/src/main.rs (hunk 3 above)
Comment (paulhauner) [reply lighthouse/src/main.rs]: Great catch!

**Comment by paulhauner** [comment]
All comments addressed! Thanks for the review @realbigsean

**Review by realbigsean — APPROVED** [review]

**Pull request merged**